pilot/pkg/proxy/envoy/discovery.go

```diff
@@ -300,6 +308,26 @@ func NewDiscoveryService(ctl model.Controller, configCache model.ConfigStoreCach
 		container.ServeMux.HandleFunc("/debug/pprof/trace", pprof.Trace)
 	}
 	out.Register(container)
+
+	if len(o.WebhookEndpoint) > 0 {
```

Comment:
nit: consider moving to a method so we can just have:

```golang
out.webhookClient = newWebhookClient(o)
```

Reply:
+1